src/optimization/steepest_descent_method.cpp

```diff
@@ -20,282 +20,282 @@ namespace optimization
+
+        lambda = GoldenSectionSearch(parameters, lambda, invert_gradiend);
+
+        vector<double> next_step_parameters(parameters.size());
+        for (size_t i = 0; i < next_step_parameters.size(); ++i)
+        {
+            next_step_parameters[i] = parameters[i] + lambda * invert_gradiend[i];
+        }
+
+        if (IsExitCriteriaFulfilled(next_step_parameters))
+        {
+            break;
+        }
 		
-		parameters = next_step_parameters;
+        parameters = next_step_parameters;
 
 		
-		if (counter == max_iteration_number_)
-		{
-			std::cout << " Minimun has not found"<<std::endl;
-			break;
-		}
-	} while (true);
-
-	return parameters;
+        if (counter == max_iteration_number_)
+        {
+            std::cout << " Minimun has not found"<<std::endl;
+            break;
+        }
+    } while (true);
+
+    return parameters;
 }
 
 bool SteepestDescentMethod::IsExitCriteriaFulfilled(
-	vector<double>& steps_array) const
+    vector<double>& steps_array) const
 {
-	double temp = 0;
-	vector<double> gradient = CalculateGradient(steps_array);
-	for (size_t i = 0; i < steps_array.size(); ++i)
-	{
-		temp += gradient[i] * gradient[i];
-	}
-	if (sqrt(temp) > precision_)
-	{
-		return false;
-	}
-	return true;
+    double temp = 0;
+    vector<double> gradient = CalculateGradient(steps_array);
+    for (size_t i = 0; i < steps_array.size(); ++i)
+    {
+        temp += gradient[i] * gradient[i];
+    }
+    if (sqrt(temp) > precision_)
+    {
+        return false;
+    }
+    return true;
 }
 
 vector<double> SteepestDescentMethod::CalculateGradient(
-	const vector<double>& parameters) const
+    const vector<double>& parameters) const
 {
-	vector<double> gradient(parameters.size());
+    vector<double> gradient(parameters.size());
 
-	for (size_t i = 0; i<parameters.size(); ++i)
-	{
-		gradient[i] = CalculateDerivative(parameters, i);
-	}
+    for (size_t i = 0; i<parameters.size(); ++i)
+    {
+        gradient[i] = CalculateDerivative(parameters, i);
+    }
 
-	return gradient;
+    return gradient;
 }
 
 double SteepestDescentMethod::CalculateDerivative(
-	const vector<double>& parameters,
-	size_t& index) const
+    const vector<double>& parameters,
+    size_t& index) const
 {
-	vector<double> decrement_function(parameters);
-	vector<double> increment_function(parameters);
-	decrement_function[index] = parameters[index] - increment_of_the_argument_ / 2;
-	increment_function[index] = parameters[index] + increment_of_the_argument_ / 2;
-	double increment_function_value = CountObjectiveFunctionValueToMinimize(increment_function);
-	double decrement_function_value = CountObjectiveFunctionValueToMinimize(decrement_function);
-	return (increment_function_value - decrement_function_value) / increment_of_the_argument_;
+    vector<double> decrement_function(parameters);
+    vector<double> increment_function(parameters);
+    decrement_function[index] = parameters[index] - increment_of_the_argument_ / 2;
```

Comment:
Don't use any magic numbers (2) in the code - make it const in anonymous namespace instead